vng_api_common/schema.py

```diff
@@ -1,177 +1,404 @@
-        self.consumes = get_consumes(api_settings.DEFAULT_PARSER_CLASSES)
-        self.produces = get_produces(api_settings.DEFAULT_RENDERER_CLASSES)
-        paths, prefix = self.get_paths(endpoints, components, request, public)
+from vng_api_common.caching.introspection import has_cache_header
+from vng_api_common.constants import HEADER_AUDIT, HEADER_LOGRECORD_ID, VERSION_HEADER
+from vng_api_common.exceptions import Conflict, Gone, PreconditionFailed
+from vng_api_common.geo import DEFAULT_CRS, HEADER_ACCEPT, HEADER_CONTENT, GeoMixin
+from vng_api_common.permissions import AuthRequired, get_required_scopes
+from vng_api_common.serializers import FoutSerializer, ValidatieFoutSerializer
 
-        security_definitions = self.get_security_definitions()
-        if security_definitions:
-            security_requirements = self.get_security_requirements(security_definitions)
-        else:
-            security_requirements = None
-
-        url = self.url
-        if url is None and request is not None:
-            url = request.build_absolute_uri()
-
-        return openapi.Swagger(
-            info=self.info,
-            paths=paths,
-            consumes=self.consumes or None,
-            produces=self.produces or None,
-            security_definitions=security_definitions,
-            security=security_requirements,
-            _url=url,
-            _prefix=prefix,
-            _version=self.version,
-            **dict(components),
-        )
-
-    def get_path_parameters(self, path, view_cls):
-        """Return a list of Parameter instances corresponding to any templated path variables.
+logger = logging.getLogger(__name__)
 
-        :param str path: templated request path
-        :param type view_cls: the view class associated with the path
-        :return: path parameters
-        :rtype: list[openapi.Parameter]
+TYPE_TO_FIELDMAPPING = {
+    OpenApiTypes.INT: serializers.IntegerField,
+    OpenApiTypes.NUMBER: serializers.FloatField,
+    OpenApiTypes.STR: serializers.CharField,
+    OpenApiTypes.BOOL: serializers.BooleanField,
+}
+
+COMMON_ERRORS = [
+    exceptions.AuthenticationFailed,
+    exceptions.NotAuthenticated,
+    exceptions.PermissionDenied,
+    exceptions.NotAcceptable,
+    Conflict,
+    Gone,
+    exceptions.UnsupportedMediaType,
+    exceptions.Throttled,
+    exceptions.APIException,
+]
+
+DEFAULT_ACTION_ERRORS = {
+    "create": COMMON_ERRORS + [exceptions.ParseError, exceptions.ValidationError],
+    "list": COMMON_ERRORS,
+    "retrieve": COMMON_ERRORS + [exceptions.NotFound],
+    "update": COMMON_ERRORS
+    + [exceptions.ParseError, exceptions.ValidationError, exceptions.NotFound],
+    "partial_update": COMMON_ERRORS
+    + [exceptions.ParseError, exceptions.ValidationError, exceptions.NotFound],
+    "destroy": COMMON_ERRORS + [exceptions.NotFound],
+}
+
+AUDIT_TRAIL_ENABLED = apps.is_installed("vng_api_common.audittrails")
+
+
+def _view_supports_audittrail(view: viewsets.ViewSet) -> bool:
```

Comment:
The comment in the [permissions](https://github.com/maykinmedia/commonground-api-common/pull/24/files#diff-a413b169bbf95056ff899e33ac0cc66f42218d5e003152974352f97eff74d3e8) files applies here too I guess